bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/AbstractArrayValue.java

```diff
@@ -151,7 +151,7 @@ public void setLength(long length) {
         int newLength = (int) length;
         checkFixedLength(length);
         rangeCheck(length, size);
-        fillerValueCheck(newLength, size);
+        fillerValueCheck(newLength, size, true);
```

Comment:
Use another method without modifying this

Reply:
fillerValueCheck is not a public API, hence modifying the existing method should be OK. what is the reason for using a different method?

Reply:
We can directly pass expectedLength here so we do not need if condition or separate method.